components/browser/toolbar/src/main/res/layout/mozac_browser_toolbar_displaytoolbar.xml

```diff
@@ -37,51 +37,73 @@
 
     <!-- URL indicators (lock, tracking protection, ..) -->
 
-    <ImageView
-        android:id="@+id/mozac_browser_toolbar_empty_indicator"
-        android:layout_width="40dp"
-        android:layout_height="40dp"
-        android:layout_marginTop="8dp"
-        android:scaleType="center"
-        android:visibility="gone"
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/mozac_browser_toolbar_indicators"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
```

Comment:
The only part I do not like that much about this PR is this nesting. We refactored the toolbar to have a flat layout and this introduces a new layer. Is this really needed or can we get the same effect inside the same ConstraintLayout?

Reply:
Yes, I don't like it much neither, the issue that I was having was the new icon it's next to the `origin_view` and when I hided it the space of the icon was still there.
 
<img width="240" alt="image" src="https://user-images.githubusercontent.com/773158/101510190-9d553380-3947-11eb-84ec-6d355603b9e9.png">


I tried with a `Barrier` that fixed the gap issue but it added a default space that I couldn't get rid of.
```xml
 <mozilla.components.browser.toolbar.display.PermissionIconView
        android:id="@+id/mozac_browser_toolbar_permission_indicator"
        android:layout_width="40dp"
        android:layout_height="40dp"
        android:layout_marginTop="8dp"
        android:importantForAccessibility="no"
        android:scaleType="center"
        android:visibility="gone"
        app:layout_constraintStart_toEndOf="@+id/mozac_browser_toolbar_security_indicator"
        app:layout_constraintTop_toTopOf="parent"
        app:srcCompat="@drawable/mozac_ic_autoplay_blocked" />

    <androidx.constraintlayout.widget.Barrier
        android:id="@+id/barrier7"
        android:layout_width="0dp"
        android:layout_height="0dp"
        app:barrierDirection="end"
        app:constraint_referenced_ids="mozac_browser_toolbar_security_indicator,mozac_browser_toolbar_permission_indicator"
         />

    <!-- URL & Title -->

    <mozilla.components.browser.toolbar.display.OriginView
        android:id="@+id/mozac_browser_toolbar_origin_view"
        android:layout_width="0dp"
        android:layout_height="40dp"
        android:layout_marginTop="8dp"
        app:layout_constraintEnd_toStartOf="@+id/mozac_browser_toolbar_page_actions"
        app:layout_constraintStart_toEndOf="@+id/barrier7"
        app:layout_constraintTop_toTopOf="parent"
        app:layout_goneMarginStart="8dp"
        app:layout_goneMarginTop="8dp" />

```

 I will give it another try to see it it could be fixed without the nesting.

Reply:
I was able used it directly without nesting, maybe when I started working on it, I was setting `INVISIBLE` instead of `GONE` not sure, why I was getting the gap. But now it's working.